usr/sbin/p11sak/p11sak.c

```diff
@@ -1142,6 +1192,152 @@ static CK_ATTRIBUTE_TYPE col2type(int col)
+                            printf(" Error in retrieving Attribute %s: %lu\n",
+                                    confignode_to_bareval(name)->value, temp.ulValueLen);
+                        } else {
+                            if (long_print) {
+                                printf("          %s: %s\n", confignode_to_bareval(name)->value,
+                                        CK_BBOOL2a(*(CK_BBOOL*) temp.pValue));
+                            } else if (strcmp(CK_BBOOL2a(*(CK_BBOOL*) temp.pValue), "CK_TRUE") == 0) {
+                                printf(" 1 ");
+                                return rc;
+                            }
+                        }    
+                    } else if (rc == CKR_ATTRIBUTE_SENSITIVE) {
+                        printf("          %s: SENSITIVE\n", confignode_to_bareval(name)->value);
+                    } else if (rc != CKR_ATTRIBUTE_TYPE_INVALID) {
+                        return rc;
+                    } 
+                } else if (strcmp((confignode_to_bareval(type)->value), "CK_ULONG") == 0) {
+                    // build template
+                    CK_ULONG a_ulong;
+                    CK_ULONG us = sizeof(CK_ULONG);
+                    CK_ATTRIBUTE temp = {hex, &a_ulong, us};
+
+                    // get attribute value
+                    rc = funcs->C_GetAttributeValue(session, hkey, &temp, 1);
+                    if (rc == CKR_OK) {
+                        if (temp.ulValueLen != sizeof(CK_ULONG)) {
+                            printf(" Error in retrieving Attribute %s: %lu\n",
+                                    confignode_to_bareval(name)->value, temp.ulValueLen);
+                        } else {
+                            if (long_print) {
+                                printf("          %s: %lu (0x%lX)\n", confignode_to_bareval(name)->value, 
+                                        *(CK_ULONG*) temp.pValue, *(CK_ULONG*) temp.pValue);    
+                            } else {
+                                printf(" 1 ");
+                                return rc;
+                            }
+                        }    
+                    } else if (rc == CKR_ATTRIBUTE_SENSITIVE) {
+                        printf("          %s: SENSITIVE\n", confignode_to_bareval(name)->value);
+                    } else if (rc != CKR_ATTRIBUTE_TYPE_INVALID) {
+                        return rc;
+                    } 
+                } else if (strcmp((confignode_to_bareval(type)->value), "CK_BYTE") == 0) {
+                    // get length 
+                    CK_ATTRIBUTE temp= {hex, NULL, 0};
+                    rc = funcs->C_GetAttributeValue(session, hkey, &temp, 1);
+
+                    if (temp.ulValueLen != CK_UNAVAILABLE_INFORMATION) {
+                        // build template
+                        int a_byte_array = temp.ulValueLen;
+                        temp.pValue = malloc(a_byte_array);
```

Comment:
Check temp.pValue for NULL, and return error if so.

Reply:
ah here is the check. I would merge this with the previous commit.

Reply:
Done